### PR TITLE
prover9: upstream Linux fixes

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Is this patch submitted upstream?

**Comment by danielnachun** [comment]
No, and this reminds me that I forgot to add a comment as to why.  The order of linker flags is not supposed to matter, but due to an idiosyncratic choice made by Ubuntu 16.04 it has to be in a certain order.  This will go away when we move to Ubuntu 18.04, after which this will patch will be not be needed.

File: Formula/prover9.rb [hunk 1]
```diff
@@ -14,6 +14,11 @@ class Prover9 < Formula
     sha256 cellar: :any_skip_relocation, high_sierra:   "055cf6646dd19effa87d7b9fa8e820c24710a023bcefc98c35604205530ab2c3"
   end
 
+  on_linux do
+    # Order of parameters passed to gcc matters
+    patch :DATA
```
Comment (carlocab) [inline Formula/prover9.rb:19]: Lets upstream this patch, please.

File: Formula/prover9.rb [hunk 2]
```diff
@@ -48,3 +56,41 @@ def install
     system bin/"mace4", "-f", testpath/"group2.in"
   end
 end
```
Comment (carlocab) [inline Formula/prover9.rb:58]:
```suggestion
end

```
Let's be consistent about formatting embedded patches.

Comment (danielnachun) [reply Formula/prover9.rb:58]: Fixed!

**Comment by danielnachun** [comment]
I added a comment explaining why we shouldn't need this patch after switching to Ubuntu 18.04 LTS.

File: Formula/prover9.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ class Prover9 < Formula
   url "https://www.cs.unm.edu/~mccune/prover9/download/LADR-2009-11A.tar.gz"
   version "2009-11A"
   sha256 "c32bed5807000c0b7161c276e50d9ca0af0cb248df2c1affb2f6fc02471b51d0"
-  license "GPL-2.0"
+  license "GPL-2.0-or-later"
```
Comment (cho-m) [inline Formula/prover9.rb:7]: I see most files are GPL-2.0-only.

e.g. `provers.src/prover9.c`:
```
    The LADR Deduction Library is free software; you can redistribute it
    and/or modify it under the terms of the GNU General Public License,
    version 2.
```

**Comment by carlocab** [comment]
We should upstream a patch anyway, even if we eventually won't need it.

**Comment by danielnachun** [comment]
I guess the other difficulty with upstreaming this is that there is no upstream VCS.  It's just a tool hosted on the author's personal website: https://www.cs.unm.edu/~mccune/prover9/.  Sadly the author also died in 2011: https://en.wikipedia.org/wiki/William_McCune.  So I'm not sure how we would go about upstreaming this.

**Review by carlocab — APPROVED** [review]
Ah, that's too bad. This is fine then.

**Comment by cho-m** [comment]
We should probably `deprecate` formula in separate PR as unmaintained.

**Comment by danielnachun** [comment]
Yeah, I'm not sure why that wasn't checked when it was added in 2019.  It will build without patching on Ubuntu 18.04 LTS, so I don't think we'll have to disable it until it doesn't build at some point in the future.

**Comment by carlocab** [comment]
I think we can skip deprecating this one until it starts causing problems/fails to build on some OS version. Deprecating formulae indefinitely can give users the wrong impression about deprecated formulae...

**Pull request merged**